core/src/software/aws/toolkits/core/utils/Waiter.kt

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.core.utils
+
+import software.amazon.awssdk.awscore.exception.AwsServiceException
+import java.time.Duration
+
+fun <T> wait(
+    call: () -> T,
+    success: (T) -> Boolean,
+    fail: (T) -> Boolean,
```

Comment:
can we combine `fail` / `errorMessage` and `failByException` / `errorMessageByException` into single parameters that accept `T`/`Exception` and return `String?` - where the return type represents the error message (`null` is considered not to be a an error).

Reply:
Done. It's always good to shrink the parameter size. :)